examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1271 @@
+              const Tensor<1, dim> &normal =
+                surface_fe_values->normal_vector(q);
+              const double error_at_point =
+                sol.at(q) - analytical_solution.value(point);
+              const Tensor<1, dim> vector_error_at_point =
+                (sol_grad.at(q) - (normal * sol_grad.at(q)) * normal -
+                 analytical_solution.gradient(point));
+
+              cell_L2_error_sqr +=
+                Utilities::pow(error_at_point, 2) * JxW_surface[q];
+              cell_H1_error_sqr +=
+                vector_error_at_point * vector_error_at_point * JxW_surface[q];
+            }
+          copy_data.cell_L2_error_sqr = cell_L2_error_sqr;
+          copy_data.cell_H1_error_sqr = cell_H1_error_sqr;
+
+          if (stabilization_scheme.needs_cell_worker())
+            {
+              typename DoFHandler<dim>::active_cell_iterator level_set_cell(
+                &(triangulation),
+                cell->level(),
+                cell->index(),
+                &level_set_dof_handler);
+              scratch_data.fe_values.reinit(cell);
+              scratch_data.level_set_fe_values.reinit(level_set_cell);
+              stabilization_scheme.evaluate_cell_worker(
+                locally_relevant_solution,
+                level_set,
+                cell,
+                scratch_data,
+                copy_data);
+            }
+        }
+    };
+
+    const auto copier = [&](const auto &copy_data) {
+      if (copy_data.cell_index < activeFE_indicator.size())
+        {
+          error_L2_sqr += copy_data.cell_L2_error_sqr;
+          error_H1_sqr += copy_data.cell_H1_error_sqr;
+          error_stab_sqr += copy_data.cell_stab_sqr;
+        }
+    };
+
+    ScratchData<dim> scratch_data(mapping,
+                                  fe_collection,
+                                  mesh_classifier,
+                                  level_set_dof_handler,
+                                  level_set,
+                                  surface_update_flags,
+                                  cell_quadrature,
+                                  quadrature_1D);
+
+    CopyData<dim> copy_data;
+
+    MeshWorker::mesh_loop(dof_handler.begin_active(),
+                          dof_handler.end(),
+                          cell_worker,
+                          copier,
+                          scratch_data,
+                          copy_data,
+                          MeshWorker::assemble_own_cells);
+    timer.stop();
+    pcout << "took (" << timer.wall_time() << "s)" << std::endl;
+  }
+
+  template <int dim>
+  void LaplaceBeltramiSolver<dim>::display_results()
+  {
+    const double error_L2 =
+      std::sqrt(Utilities::MPI::sum(error_L2_sqr, mpi_communicator));
+    const double error_semiH1 =
+      std::sqrt(Utilities::MPI::sum(error_H1_sqr, mpi_communicator));
+    const double error_stab =
+      std::sqrt(Utilities::MPI::sum(error_stab_sqr, mpi_communicator));
```

Comment:
You will want to do this in the previous function so that at the end of the function, the variables that contain this information are (globally) correct.